modules/transactions/src/main/java/org/apache/ignite/internal/tx/storage/state/rocksdb/TxStateRocksDbStorage.java

```diff
@@ -57,6 +61,24 @@
  * Tx state storage implementation based on RocksDB.
  */
 public class TxStateRocksDbStorage implements TxStateStorage {
+    private static final VarHandle STATE;
+
+    private static final VarHandle REBALANCE_FUTURE;
+
+    static {
+        try {
+            STATE = MethodHandles.lookup().findVarHandle(TxStateRocksDbStorage.class, "state", StorageState.class);
```

Comment:
What's the problem with using atomics?
Are you sure that placing these values inside of the object is beneficial? Do you have proofs?
I'd prefer a code simplicity if the advantage is not clear.

Reply:
Since there can be many partitions, we can, although a little, win in memory consumption for each using `VarHandle` instead of atomics. The less memory is used, the faster the GC runs, the faster the system runs.

Yes, it looks like premature optimization, but I don't use complex constructs to make the code much more complicated.

Reply:
Ok, I give up. Generally speaking, less code is better than more code. Benefits of the memory saved is not clear at all.